packages/ra-ui-materialui/src/layout/Title.tsx

```diff
@@ -23,6 +23,10 @@ const Title: FC<TitleProps> = ({
         typeof document !== 'undefined'
             ? document.getElementById('react-admin-title')
             : null;
+
+    const documentTitle = typeof title === 'string' ? title : defaultTitle;
+    useDocumentTitle(translate(documentTitle, { _: documentTitle }));
```

Comment:
I think you shouldn't set the document title when the titleProp is an element. In that case, it's the responsibility of the element to define the document title. 

Also, the defaultTitle isn't translated on line 35, but in your case it is. This will lead to different document and page title.